src/ExportMap.js

```diff
@@ -406,7 +413,8 @@ ExportMap.parse = function (path, content, context) {
     },
   });
 
-  if (!unambiguous.isModule(ast) && !hasDynamicImports) return null;
+  const maybeNotEsm = !unambiguous.isModule(ast);
+  if (maybeNotEsm && !hasDynamicImports) return null;
```

Comment:
it could still be ESM in this case - it depends on what the eslint config has set for `sourceType`.

Reply:
Do you suggest adding a check on the `sourceType` value in the config? I don't see any code checking this at the moment, but maybe I am not looking for it correctly.

Reply:
You're right, we probably aren't checking that. I think that specifically when the goal is ambiguous, we should defer to the config (and add that checking)